### PR TITLE
Parser for Nion Co. nd binary and HDF5 files to NeXus, fixes #27

### LINKED ISSUE #27: Add parsing for nionswift
- [x] Generalized store data for all cases that were observed in Christoph's lab from the Nion data from the last 5 years
- [x] Generalized NXdata for all use cases from the above-mentioned collection
- [x] Generalized mapping table for all metadata from majority of unique concepts from the above-mentioned collection
- [x] Testing with old example dataset
- [x] Update pynxtools/sprint23_em_v3 with latest versions for image and spectrum
- [x] Final testing with metadata mapping

### PR DESCRIPTION
**Pull request by mkuehbach**

### CODE REVIEW

File: src/pynxtools_em/parsers/nxs_nion.py [hunk 1]
```diff
@@ -431,3 +436,118 @@ def parse(self, template: dict) -> dict:
                 print("Parsing in-place nionswift project (nsproj + data)...")
             self.parse_project_file(template)
         return template
+
+    def process_em_data(self, template: dict) -> dict:
+        # TODO
+        dimensional_calibrations = None
+        shape = None
+        # [{'offset': -0.022, 'scale': 0.0013333333333333333, 'units': '1/nm'},
```
Comment (lukaspie) [inline src/pynxtools_em/parsers/nxs_nion.py:444]: Remove if not needed anymore.

File: src/pynxtools_em/configurations/nion_cfg.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    ],
+    "map_to_f8": [
+        ("LENS_EM[lens1]/value", "C1 ConstW"),
+        ("LENS_EM[lens2]/value", "C2 ConstW"),
+        ("LENS_EM[lens3]/value", "C3 ConstW"),
+    ],
+}
+
+UNITS_TO_SPECTRUM_LIKE_NXDATA = {
+    "eV": ("spectrum_0d", ["energy"]),
```
Comment (lukaspie) [inline src/pynxtools_em/configurations/nion_cfg.py:132]: I would expect that keys and values here are reversed. I.e., if there is a `spectrum_0d` object, it should get unit `eV`. But I don't know enough of this code to say what makes sense here.

**Review by lukaspie — COMMENTED** [review]
Can't comment much, but the configurations dicts seems a bit sketchy to me. The semicolon-seperated keys (like `"unitless;nm;nm;nm"`) don't look very pythonic.

File: src/pynxtools_em/parsers/nxs_nion.py (hunk 1 above)
Comment (mkuehbach) [reply src/pynxtools_em/parsers/nxs_nion.py]: Implemented

**Comment by mkuehbach** [comment]
> Can't comment much, but the configurations dicts seems a bit sketchy to me. The semicolon-seperated keys (like `"unitless;nm;nm;nm"`) don't look very pythonic.

Good point changed to "_"

**Pull request merged**